core/src/main/java/com/sequenceiq/cloudbreak/service/credential/CredentialService.java

```diff
@@ -73,10 +76,8 @@
     }
 
     public Credential get(Long id) {
-        Credential credential = credentialRepository.findOne(id);
-        if (credential == null) {
-            throw new AccessDeniedException(String.format("Access is denied: Credential '%d'.", id));
-        }
+        Optional<Credential> credentialOpt = credentialRepository.findById(id);
+        Credential credential = credentialOpt.orElseThrow(() -> new AccessDeniedException(String.format("Access is denied: Credential '%d'.", id)));
```

Comment:
This looks like an error, if credential not found would be better to throw NotFoundEception.

Reply:
I think this way the response code (403) is better from a security point of view.

Reply:
Response code on the rest endpoint and logged exception is two different thing. We have to log NotFoundException and response 403.

Reply:
@bbihari and other developers are currently working on a feature that will standardise access handling. I'll revert this part of my change to the original code, since this PR is about upgrading project dependencies with the aim of maintaining the original behaviour.

Reply:
Ok but please inform them about this wrong behavior, because they need to fix it.